jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/go/GoLambdaHandlerResolver.kt

```diff
@@ -3,23 +3,84 @@
 
 package software.aws.toolkits.jetbrains.services.lambda.go
 
+import com.goide.psi.GoFunctionDeclaration
+import com.goide.psi.GoTokenType
+import com.goide.psi.GoType
+import com.goide.psi.GoTypeList
+import com.goide.psi.impl.GoLightType
+import com.goide.psi.impl.GoPsiUtil
+import com.goide.stubs.index.GoFunctionIndex
+import com.goide.stubs.index.GoIdFilter
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.psi.NavigatablePsiElement
 import com.intellij.psi.PsiElement
 import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.util.elementType
 import software.aws.toolkits.jetbrains.services.lambda.LambdaHandlerResolver
 
 class GoLambdaHandlerResolver : LambdaHandlerResolver {
-    override fun findPsiElements(project: Project, handler: String, searchScope: GlobalSearchScope): Array<NavigatablePsiElement> {
-        TODO("Not yet implemented")
-    }
+    override fun findPsiElements(project: Project, handler: String, searchScope: GlobalSearchScope): Array<NavigatablePsiElement> =
+        // GoFunctionDeclarationImpl is a NavigatablePsiElement
+        GoFunctionIndex.find(handler, project, searchScope, GoIdFilter.getFilesFilter(searchScope))
+            .filterIsInstance<GoFunctionDeclaration>()
+            .filter { it.isValidHandlerIdentifier() }
+            .filterIsInstance<NavigatablePsiElement>()
+            .toTypedArray()
 
     override fun determineHandler(element: PsiElement): String? {
-        TODO("Not yet implemented")
```

Comment:
readAction needed?

Reply:
We don't seem to do that in python/other runtimes?

Reply:
I believe it is not needed if invoked from EDT?